src/fstab-generator/fstab-generator.c

```diff
@@ -112,7 +111,7 @@ static int add_swap(
                 return 0;
         }
 
-        if (detect_container() > 0) {
+        if (path_is_read_only_fs("/sys") > 0) {
```

Comment:
Please describe the motivation of such changes in the commit message…

Is this for speed? I think it'd be more reasonable to do `if (path_is_read_only_fs("/sys") > 0 || detect_container() > 0) …`. As the change is done now, effectively fstab-generator could execute where it didn't before, in case of containers that are set up with /sys writeable. It might be an error in the set up, or just a different way to do things, and I don't think we want to open ourselves to regressions here.

Reply:
We replaced `detect_container()` -> `path_is_read_only_fs("/sys")` in networkd. So, I thought that we should always use `path_is_read_only_fs("/sys")` everywhere. But after reconsidering about that, maybe we should use it only when checking udevd is running or not. In this case, swap is not related to udevd. So, I will drop the change.